infullmvp-kotlin/src/main/java/com/infullmobile/android/infullmvp/LaunchableIntent.kt

```diff
@@ -2,10 +2,19 @@ package com.infullmobile.android.infullmvp
 
 import android.app.Activity
 import android.content.Intent
+import com.infullmobile.android.infullmvp.requestcodes.FibonacciRequestCodes
+import com.infullmobile.android.infullmvp.requestcodes.RequestCodeProvider
 
 open class LaunchableIntent(private val activity: Activity,
                             private val intent: Intent) {
 
+    open var requestCodeProvider: RequestCodeProvider<String> = DEFAULT_REQUEST_CODE_PROVIDER
+
+    private val innerProvider: RequestCodeProvider<String>
+        get() = requestCodeProvider
+
+    val requestCode: Int = innerProvider.requestCodeFor(this.intent.component.className)
```

Comment:
@bhm  Launchable intent has a state now, which should be persisted in case the Activity which started another for result becomes destroyed. I am not sure whether this feature is worth the hassle.